certbot/certbot/_internal/renewal.py

```diff
@@ -199,9 +199,9 @@ def restore_required_config_elements(config: configuration.NamespaceConfig,
         zip(INT_CONFIG_ITEMS, itertools.repeat(_restore_int)),
         zip(STR_CONFIG_ITEMS, itertools.repeat(_restore_str)))
     for item_name, restore_func in required_items:
-        if item_name in renewalparams and not cli.set_by_cli(item_name):
+        if item_name in renewalparams and not config.set_by_user(item_name):
             value = restore_func(item_name, renewalparams[item_name])
-            setattr(config.namespace, item_name, value)
```

Comment:
All the times we explicitly called `config.namespace` were unnecessary before, right? Since we overrode get and set?

Reply:
that's my understanding, yeah

Reply:
:facepalm: